### PR TITLE
Consider virtual builders as active

### PR DESCRIPTION
**Pull request by rajdeepbharati**
Fixes https://github.com/buildbot/buildbot/issues/4885

## Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: www/base/src/app/builders/builders.controller.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ class Builders {
                     active = true;
                 }
             }
+            // virtual builders need to be supplied 'virtual' tag
+            // to consider them as active builders
+            if (builder.tags.includes('virtual')) {
```
Comment (tardyp) [inline www/base/src/app/builders/builders.controller.js]: I would reserve that keyword, using '_virtual_'.
Also, we need to make sure '_virtual_' tag is set for virtual builders
https://github.com/buildbot/buildbot/blob/e8b3696a23335236b69e5ebaa7d7976c5d994f2a/master/buildbot/process/build.py#L271

Comment (rajdeepbharati) [reply www/base/src/app/builders/builders.controller.js]: Done

File: master/buildbot/process/build.py [hunk 2]
```diff
@@ -267,6 +267,8 @@ def getBuilderId(self):
                 tags = self.getProperty(
                     self.VIRTUAL_BUILDERTAGS_PROP,
                     self.builder.config.tags)
+                if '_virtual_' not in tags:
```
Comment (tardyp) [inline master/buildbot/process/build.py]: you wll need to update the unit tests for this code

Comment (rajdeepbharati) [reply master/buildbot/process/build.py]: Updated

**Pull request merged**